checks/branch_protection.go

```diff
@@ -109,6 +128,198 @@ func BranchProtection(c *checker.CheckRequest) checker.CheckResult {
 	return checkReleaseAndDevBranchProtection(c.RepoClient, c.Dlogger)
 }
 
+func updateMaxScore(s1, s2 int) (int, error) {
+	if s2 != 0 && s1 != 0 && s1 != s2 {
```

Comment:
I think I'm misunderstanding this logic :/ why is it the case that across all branches the settings are the same or one is 0? 

Is it because the scores are binary (at least for now, given the values in `branchProtectionSettingScores`)?

Reply:
the name of the function may be confusing. This is not the result score: it's the maximum score possible for a level, calculated by adding `+1` for each settings enabled. The maximum score depends on whether you have admin access or not. When you don't have admin access, the max score for a level may be 0 because the settings are not available. I don''t want to say "we've removed points because we cannot access your admin-only settings".
So in this function, I update the maximum possible score but check that it's not 0, ie that is it available.
As yoou say, the maximum score across all branches MUST be the same: I added this code to also verify that the maximum possible score is the same across all branches: if this is violated it means there's a bug in our implementation.
Does this make sense?

Reply:
I think you're right there is unecessary logic I forgot to remove. let me update

Reply:
done, PTAL

Reply:
Ah! Gotcha. Was just a verification /debug check. Got it with the removed logic. Thank you!